internal/app/go-filecoin/paymentchannel/manager.go

```diff
@@ -261,38 +344,63 @@ func (pm *Manager) createPaymentChannelWithVoucher(paychAddr address.Address, vo
 
 // saveNewVoucher saves a voucher to an existing payment channel
 func (pm *Manager) saveNewVoucher(paychAddr address.Address, voucher *paychActor.SignedVoucher, proof []byte) error {
-	var chinfo ChannelInfo
-	st := pm.paymentChannels.Get(paychAddr)
-	if err := st.Get(&chinfo); err != nil {
+	has, err := pm.paymentChannels.Has(paychAddr)
+	if err != nil {
 		return err
 	}
-	if chinfo.NextLane <= voucher.Lane {
-		return xerrors.Errorf("lane does not exist %d", voucher.Lane)
-	}
-	if chinfo.HasVoucher(voucher) {
-		return xerrors.Errorf("voucher already saved")
+	if !has {
+		return xerrors.Errorf("channel does not exist %s", paychAddr.String())
 	}
-	if err := pm.paymentChannels.
-		Get(paychAddr).
-		Mutate(func(info *ChannelInfo) error {
-			info.NextNonce++
-			info.Vouchers = append(info.Vouchers, &VoucherInfo{
-				Voucher: voucher,
-				Proof:   proof,
-			})
-			return nil
-		}); err != nil {
+	if err := pm.paymentChannels.Mutate(paychAddr, func(info *ChannelInfo) error {
+		if info.NextLane <= voucher.Lane {
+			return xerrors.Errorf("lane does not exist %d", voucher.Lane)
+		}
+		if info.HasVoucher(voucher) {
+			return xerrors.Errorf("voucher already saved")
+		}
+		info.NextNonce++
+		info.Vouchers = append(info.Vouchers, &VoucherInfo{
+			Voucher: voucher,
+			Proof:   proof,
+		})
+		return nil
+	}); err != nil {
 		return err
 	}
 	return nil
 }
 
-// GetMinerWorkerAddress mocks getting a miner worker address from the miner address
-func (pm *Manager) GetMinerWorkerAddress(ctx context.Context, miner address.Address, tok shared.TipSetToken) (address.Address, error) {
-	view, err := pm.stateViewer.GetStateView(ctx, tok)
-	if err != nil {
-		return address.Undef, err
-	}
-	_, fcworker, err := view.MinerControlAddresses(ctx, miner)
-	return fcworker, err
+//  paychStore is a thin threadsafe wrapper for StateStore
+type paychStore struct {
+	storeLk sync.RWMutex
+	store   *statestore.StateStore
+}
+
+type mutator func(info *ChannelInfo) error
+
+func (ps *paychStore) Mutate(addr address.Address, m mutator) error {
+	ps.storeLk.Lock()
+	defer ps.storeLk.Unlock()
+	return ps.store.Get(addr).Mutate(m)
+}
+func (ps *paychStore) List(info *[]ChannelInfo) error {
+	ps.storeLk.Lock()
```

Comment:
This and the next 2 methods should use `ps.storeLk.RLock()` and `ps.storkLk.RUnlock()` (unless they actually mutate state).